### ISSUE TITLE
Bounty RTL8720DN

### ISSUE THREAD

**Issue by nx10** [kuba2k2/libretuya#1]
Cool project. I hope it's ok for me to add this as an issue, feel free to close it.

In case you don't know there is a $ 100 bounty on platformio RTL8720DN support:

https://github.com/platformio/platformio-core/issues/769#issuecomment-903447789

**Comment by kuba2k2** [comment]
Unfortunately, I do not own any RTL87xxD chips. As they're substantially different, I can't guarantee that my current code will work on them. Though if this project gets more interest, I will consider buying one of these boards in the future.

**Comment by kuba2k2** [comment]
Right, I see. I'm not entirely sure if my project would suffice for their needs: because Realtek's SDK and Arduino cores are quite messy, I'm mostly rewriting everything from scratch to make it compatible with ESP32 cores. 

While this extends the possibilities of running existing ESP32 programs, it makes the code less compatible with non-PlatformIO Arduino cores - i.e. those from Realtek itself. That's why I'm not sure if this is what the community wants to have.

Even then, I'm still planning to do AmebaD support at some point - after all, the base code is not that much different from AmebaZ which I have already ported. Currently, this allowed me to run ESPHome with minimal modifications on RTL8710BN.

If you're interested in using my project with RTL8720D, and would like to help - let me know, as any contribution will be much appreciated.

**Comment by BinaryBrain86** [comment]
Hi, I appreciate you work and effort in this project. I followed the conversation on [elektroda](https://www.elektroda.com/rtvforum/viewtopic.php?p=19984465#19984465) and contributed to his project another device I own. That's how I found and saw this project.

I own now two other Tuya devices (power switch) with the [WR2E Module](https://developer.tuya.com/en/docs/iot/wr2e?id=K97scnsjhue4h), which have the Realtek RTL8710BN MCU. 

> Currently, this allowed me to run ESPHome with minimal modifications on RTL8710BN.

I'm wondering If you have kind of a tutorial or some small steps how to do this.
Thanks in advanced allot for reading and writing back.

**Comment by kuba2k2** [comment]
Yes - we have an entire small community :slightly_smiling_face: 
We usually discuss everything on Discord and GitHub.

There's a getting-started guide: https://docs.libretuya.ml/docs/getting-started/

Also a guide for Realtek dumping/flashing: https://docs.libretuya.ml/docs/platform/realtek-ambz/dumping/

**Comment by BinaryBrain86** [comment]
> Yes - we have an entire small community 🙂 We usually discuss everything on Discord and GitHub.
> 
> There's a getting-started guide: https://docs.libretuya.ml/docs/getting-started/
> 
> Also a guide for Realtek dumping/flashing: https://docs.libretuya.ml/docs/platform/realtek-ambz/dumping/

Thank you, I'll check it out.

**Comment by MrAppAndCrap** [comment]
> Yes - we have an entire small community 🙂 We usually discuss everything on Discord and GitHub.
> 
> There's a getting-started guide: https://docs.libretuya.ml/docs/getting-started/
> 
> Also a guide for Realtek dumping/flashing: https://docs.libretuya.ml/docs/platform/realtek-ambz/dumping/

Hey, what is the discord link / name? Thanks

**Comment by kuba2k2** [comment]
Hi, it's at the top of the README.

**Comment by wrobelda** [comment]
Actually, am I wrong or is this the SDK? https://github.com/Ameba-AIoT/ameba-rtos-d

**Comment by Bl00d-B0b** [comment]
A working AmebaD port is up for review in #403: ESPHome running on an Ai-Thinker BW16-Kit (RTL8720DX, the USB-C dev board — https://www.amebaiot.com.cn/en/amebad-bw16-arduino-getting-started/) — WiFi STA, native API, OTA in both slots, PWM/GPIO outputs, preferences — with a bench-verified status table in the PR docs. Flashing works today with BK7231GUIFlashTool (full-image at 0x0); ltchiptool support is drafted in libretiny-eu/ltchiptool#98. Testers with BW16/RTL8720DN hardware welcome — the more boards this runs on before merge, the better.
